examples/webrtc/main.go

```diff
@@ -99,24 +99,25 @@ func main() {
 		}
 	}
 
-	// Set the remote SessionDescription
-	err = peerConnection.SetRemoteDescription(offer)
-	if err != nil {
-		panic(err)
-	}
-
 	// Create an answer
 	answer, err := peerConnection.CreateAnswer(nil)
 	if err != nil {
 		panic(err)
 	}
 
-	// Sets the LocalDescription, and starts our UDP listeners
-	err = peerConnection.SetLocalDescription(answer)
+	// Create channel that is blocked until ICE Gathering is complete (non-trickle ICE)
+	  gatherComplete := webrtc.GatheringCompletePromise(peerConnection)
+	  peerConnection.SetLocalDescription(answer)
+	  <-gatherComplete
+	
+	// Set the remote SessionDescription
+	err = peerConnection.SetRemoteDescription(offer)
 	if err != nil {
 		panic(err)
 	}
 
+
+
 	// Output the answer in base64 so we can paste it in browser
 	fmt.Println(signal.Encode(answer))
```

Comment:
This should be:

```go
<-webrtc.GatheringCompletePromise(peerConnection)
fmt.Println(signal.Encode(*peerConnection.LocalDescription()))
```

Reply:
Cool! I edited the code

Reply:
Since we are only initiating createAnswer to start gathering ICE candidates (I think that’s right) I went ahead and set that equal to a blank identifier.

Reply:
The ICE gathering starts in SetLocalDescription. We need SetRemoteDescription to know whose side that's controlling.